### ISSUE TITLE
Variables passed to `--env` with an empty value are incorrectly parsed

### ISSUE THREAD

**Issue by benjaminblack** [webpack/webpack-cli#3284]
**Describe the bug**

When passing env vars to webpack cli using `--env`, if the value of the variable passed is empty, then the resulting variable name and value is parsed incorrectly. As an example, `webpack --env TEST=""` results in the following `env`: 

```
{
  WEBPACK_BUNDLE: true,
  WEBPACK_BUILD: true,
  'TEST=': true
}
```

**What is the current behavior?**

Empty variables passed to `--env` like `--env VAR=""` or `--env VAR=$NOT_DEFINED` result in the `env` object containing property names like `VAR=` with  the value `true`.

**To Reproduce**

Minimal webpack.config.js demonstrating behavior:

```js
import { fileURLToPath } from "node:url";

export default (env) => {
  console.log(env);

  return {
    mode: "production",
    entry: "./main.js",
    output: {
      path: fileURLToPath(new URL(".", import.meta.url)),
      filename: "main.bundle.js",
    },
  };
};
```

1.
```
$ npx webpack --env TEST1="" --env TEST2=$NON_EXISTENT_ENVVAR --env TEST3=
```

**Expected behavior**

Empty variables could be represented as one of `null`, `undefined`, or `""`, but certainly should not be `true`, and the property name should not include the `=`.

**Screenshots**

N/A

**Please paste the results of `npx webpack-cli info` here, and mention other relevant information**

```bash
npx webpack --env TEST1="" --env TEST2=$NON_EXISTENT_ENVVAR --env TEST3=
{
  WEBPACK_BUNDLE: true,
  WEBPACK_BUILD: true,
  'TEST1=': true,
  'TEST2=': true,
  'TEST3=': true
}
asset main.bundle.js 50 bytes [compared for emit] [minimized] (name: main)
./main.js 28 bytes [built] [code generated]
webpack 5.73.0 compiled successfully in 82 ms
```

**Additional context**

None

**Comment by alexander-akait** [comment]
PR welcome

**Comment by snitin315** [comment]
But yes I feel if it ends with `=` we should set it to undefined instead of setting `foo=: true`. I'll raise a patch for this.

**Comment by benjaminblack** [comment]
I'm not invoking Webpack through the terminal, I'm doing it through package.json.

With this script:
```
 "webpack": "webpack --env TEST=\"\""
```

The result is a property of `TEST=` with a value of `true`.

With this:

```
  "webpack": "webpack --env TEST=\\\"\\\""
```

The result is a property of `TEST`, but the value is `'""'` -- which is not an empty string, which would be falsy, but a string containing `""`, which is truthy.

No matter which variation I try, I can't get a property of `TEST` with a value that is falsy, which is what I'd expect.

**Comment by alexander-akait** [comment]
> The result is a property of TEST, but the value is '""' -- which is not an empty string, which would be falsy, but a string containing "", which is truthy.

Expected, because you have `""` string, it is truthy

But `webpack --env TEST=""` should be falsy, it is bug on our side

**Comment by benjaminblack** [comment]
Right, I was just trying to replicate, within a package.json script, the same behavior seen in a command-line shell running `webpack --env TEST=\"\"`, and noting that I couldn't, with either `\"\"` or with `\\\"\\\"`.
